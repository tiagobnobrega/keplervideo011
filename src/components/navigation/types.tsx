/*
 * Copyright (c) 2024 Amazon.com, Inc. or its affiliates.  All rights reserved.
 *
 * PROPRIETARY/CONFIDENTIAL.  USE IS SUBJECT TO LICENSE TERMS.
 */

import { DrawerScreenProps } from '@amzn/react-navigation__drawer/lib/typescript/src/types';
import { StackNavigationProp } from '@amzn/react-navigation__stack';
import { StackScreenProps } from '@amzn/react-navigation__stack/lib/typescript/src/types';
import { ImageSourcePropType, TouchableOpacity } from 'react-native';
import { TitleData } from '../../types/TitleData';

export enum Screens {
  DEFAULT_SCREEN = 'Home',
  APP_DRAWER = 'AppDrawer',
  HOME_SCREEN = 'Home',
  DETAILS_SCREEN = 'Details',
  PLAYER_SCREEN = 'Player',
  SETTINGS_SCREEN = 'Settings',
  SEARCH_SCREEN = 'Search',
  SEARCH_RESULTS_SCREEN = 'SearchResultsScreen',
  FEEDBACK_SCREEN = 'FeedBackScreen',
}

export enum DrawerType {
  PERMANENT = 'permanent',
  FRONT = 'front',
  SLIDE = 'slide',
  BACK = 'back',
}

export type AppStackParamList = {
  [Screens.APP_DRAWER]: undefined;
  [Screens.HOME_SCREEN]: undefined;
  [Screens.SEARCH_SCREEN]: undefined;
  [Screens.SETTINGS_SCREEN]: undefined;
  [Screens.FEEDBACK_SCREEN]: undefined;
  [Screens.SEARCH_RESULTS_SCREEN]: {
    searchKeyword: string;
  };
  [Screens.DETAILS_SCREEN]: {
    data: TitleData;
    sendDataOnBack: (id: number | string) => void;
  };
  [Screens.PLAYER_SCREEN]: {
    data: TitleData;
    sendDataOnBack: () => void;
  };
};

export type AppStackScreenProps<T extends keyof AppStackParamList> =
  StackScreenProps<AppStackParamList, T>;

export type HomeScreenNavigationProps = StackNavigationProp<
  AppStackParamList,
  Screens.HOME_SCREEN
>;

export type AppDrawerParamList = {
  [Screens.HOME_SCREEN]: undefined;
  [Screens.SETTINGS_SCREEN]: undefined;
  [Screens.SEARCH_SCREEN]: undefined;
};

export type AppDrawerScreenProps<T extends keyof AppDrawerParamList> =
  DrawerScreenProps<AppDrawerParamList, T>;

declare global {
  namespace ReactNavigation {
    interface RootParamList extends AppStackParamList {}
  }
}

export interface ButtonConfig {
  onPress: () => void;
  image: ImageSourcePropType;
  label: string;
  ref?: React.RefObject<TouchableOpacity>;
  testID: string;
}
