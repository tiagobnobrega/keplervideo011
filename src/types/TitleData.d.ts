import { MediaItem } from '@amzn/keplerblocks-types';

export type MimeType =
  | 'text/plain'
  | 'text/vtt'
  | 'application/x-subtitle' // .srt
  | 'application/x-subtitle-vtt' // .vtt
  | 'application/x-subtitle-sami'
  | 'application/x-subtitle-tmplayer'
  | 'application/x-subtitle-mpl2'
  | 'application/x-subtitle-dks'
  | 'application/x-subtitle-qttext'
  | 'application/x-subtitle-lrc';

export type VideoFormat = 'MPD' | 'HLS' | 'MP4' | 'DASH';

interface DrmScheme {
  name: string;
  licenseUri: string;
  headerTag?: string;
  headerData?: string;
}
interface TextTrack {
  label: string;
  language: string;
  uri: string;
  mimeType: MimeType;
}

export interface TitleData extends MediaItem {
  videoUrl: string;
  categories: string[];
  channelID: string;
  posterUrl: string;
  format: string;
  drmScheme?: DrmScheme;
  textTrack?: TextTrack[];
  uhd: boolean;
  secure: boolean;
  aCodec?: string;
  vCodec?: string;
  rentAmount: string;
}
