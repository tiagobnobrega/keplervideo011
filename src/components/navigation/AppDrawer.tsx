/*
 * Copyright (c) 2024 Amazon.com, Inc. or its affiliates.  All rights reserved.
 *
 * PROPRIETARY/CONFIDENTIAL.  USE IS SUBJECT TO LICENSE TERMS.
 */
import {
  createDrawerNavigator,
  DrawerContentComponentProps,
} from '@amzn/react-navigation__drawer';
import React, { useCallback } from 'react';
import { StyleSheet } from 'react-native';
import HomeSvg from '../../assets/svgr/HomeSVG';
import SearchSvg from '../../assets/svgr/SearchSVG';
import SettingSvg from '../../assets/svgr/SettingSVG';
import HomeScreen from '../../screens/HomeScreen';
import { COLORS } from '../../styles/Colors';
import { scaleUxToDp } from '../../utils/pixelUtils';
import DrawerContent from './navigationDrawerContent/DrawerContent';
import { AppDrawerParamList, DrawerType, Screens } from './types';

const SearchScreen = React.lazy(() => import('../../screens/SearchScreen'));
const SettingsScreen = React.lazy(() => import('../../screens/SettingsScreen'));
const Drawer = createDrawerNavigator<AppDrawerParamList>();

const AppDrawer = () => {
  const renderDrawerContent = useCallback(
    (props: DrawerContentComponentProps) => <DrawerContent {...props} />,
    [],
  );

  const navigationOptions = {
    drawerType: DrawerType.PERMANENT,
    drawerStyle: styles.drawer,
    headerShown: false,
  };

  return (
    <Drawer.Navigator
      screenOptions={navigationOptions}
      initialRouteName={Screens.DEFAULT_SCREEN}
      drawerContent={renderDrawerContent}>
      <Drawer.Screen
        name={Screens.HOME_SCREEN}
        component={HomeScreen}
        options={{
          // eslint-disable-next-line react/no-unstable-nested-components
          drawerIcon: props => (
            <HomeSvg
              testID="homesvg"
              width={40}
              height={40}
              stroke={COLORS.SMOKE_WHITE}
              fill={props.color}
            />
          ),
        }}
      />
      <Drawer.Screen
        name={Screens.SEARCH_SCREEN}
        component={SearchScreen}
        options={{
          // eslint-disable-next-line react/no-unstable-nested-components
          drawerIcon: () => (
            <SearchSvg
              testID="searchsvg"
              width={40}
              height={40}
              stroke={COLORS.SMOKE_WHITE}
              fill={COLORS.SMOKE_WHITE}
            />
          ),
        }}
      />
      <Drawer.Screen
        name={Screens.SETTINGS_SCREEN}
        component={SettingsScreen}
        options={{
          // eslint-disable-next-line react/no-unstable-nested-components
          drawerIcon: props => (
            <SettingSvg
              testID="settingsvg"
              width={40}
              height={40}
              stroke={COLORS.SMOKE_WHITE}
              fill={props.color}
            />
          ),
        }}
      />
    </Drawer.Navigator>
  );
};
export default React.memo(AppDrawer);

const styles = StyleSheet.create({
  drawer: {
    backgroundColor: COLORS.TRANSPARENT,
    borderRightColor: COLORS.TRANSPARENT,
    width: scaleUxToDp(100),
  },
  imageIcon: {
    width: scaleUxToDp(30),
    height: scaleUxToDp(30),
    tintColor: COLORS.SMOKE_WHITE,
  },
});
