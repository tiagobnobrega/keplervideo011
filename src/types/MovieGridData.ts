import { TitleData } from './TitleData';

export type MovieGridData = {
  heading: string;
  testID: string;
  data: () => TitleData[];
};
