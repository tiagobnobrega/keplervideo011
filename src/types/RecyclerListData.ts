import { StyleProp, ViewStyle } from 'react-native';
import { RecyclerListViewProps } from 'recyclerlistview';

export interface CustomRecyclerListViewProps extends RecyclerListViewProps {
  contentContainerStyle?: StyleProp<ViewStyle>;
  initialNumToRender: number | undefined;
}
