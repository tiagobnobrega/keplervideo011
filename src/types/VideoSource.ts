// adapted from GStreamer MIME Types (https://gstreamer.freedesktop.org/documentation/subparse/subparse.html?gi-language=c)
export type MimeType =
  | 'text/plain'
  | 'text/vtt'
  | 'application/x-subtitle' // .srt
  | 'application/x-subtitle-vtt' // .vtt
  | 'application/x-subtitle-sami'
  | 'application/x-subtitle-tmplayer'
  | 'application/x-subtitle-mpl2'
  | 'application/x-subtitle-dks'
  | 'application/x-subtitle-qttext'
  | 'application/x-subtitle-lrc';

export type VideoFormat = 'MPD' | 'HLS' | 'MP4' | 'DASH';

interface DrmScheme {
  name: string;
  licenseUri: string;
  headerTag?: string;
  headerData?: string;
}
interface TextTrack {
  label: string;
  language: string;
  uri: string;
  mimeType: MimeType;
}

export interface VideoSource {
  title: string;
  uri: string;
  format: VideoFormat;
  drmScheme?: DrmScheme;
  textTrack?: TextTrack[];
  uhd: boolean;
  secure: boolean;
  aCodec?: string;
  vCodec?: string;
}
