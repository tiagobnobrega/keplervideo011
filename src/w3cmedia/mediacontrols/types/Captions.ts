import { VideoPlayer } from '@amzn/react-native-w3cmedia';

export interface CaptionButtonProps {
  onPress: () => void;
  video: VideoPlayer;
  captionMenuVisibility: boolean;
  testID: string;
  playerControlType?: string;
  isCaptionButtonFocused: boolean | null;
}

export interface CaptionOptionProps {
  video: VideoPlayer;
  enableCaption: (captionId: string) => void;
  selectedCaptionId: string | null;
}

export interface CaptionMenuItemProps {
  text: string;
  index?: string;
  selected?: boolean;
  onPress: () => void;
  testID?: string;
}

export interface CaptionMenuProps {
  captionMenuVisibility: boolean;
  video: VideoPlayer;
  setSelectedCaption: (id: string) => void;
}
