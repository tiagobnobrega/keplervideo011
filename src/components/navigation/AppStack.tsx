/*
 * Copyright (c) 2024 Amazon.com, Inc. or its affiliates.  All rights reserved.
 *
 * PROPRIETARY/CONFIDENTIAL.  USE IS SUBJECT TO LICENSE TERMS.
 */
import { createStackNavigator } from '@amzn/react-navigation__stack';
import React, { useEffect } from 'react';
import { useDispatch } from 'react-redux';
import { AppDispatch } from '../../store';
import { fetchPlaylists } from '../../store/search/searchSlice';
import AppDrawer from './AppDrawer';
import { AppStackParamList, Screens } from './types';

const DetailsScreen = React.lazy(() => import('../../screens/DetailsScreen'));
const PlayerScreen = React.lazy(() => import('../../screens/PlayerScreen'));
const SearchResultsScreen = React.lazy(
  () => import('../../screens/SearchResultsScreen'),
);
const FeedBackScreen = React.lazy(() => import('../../screens/FeedbackScreen'));
const Stack = createStackNavigator<AppStackParamList>();
const AppStack = () => {
  const dispatch = useDispatch<AppDispatch>();

  useEffect(() => {
    //fetching local file data for search flow
    dispatch(fetchPlaylists());
  }, [dispatch]);

  const navigationOptions = {
    headerShown: false,
    animationEnabled: false,
  };

  return (
    <Stack.Navigator screenOptions={navigationOptions}>
      <Stack.Screen name={Screens.APP_DRAWER} component={AppDrawer} />
      <Stack.Screen name={Screens.DETAILS_SCREEN} component={DetailsScreen} />
      <Stack.Screen name={Screens.PLAYER_SCREEN} component={PlayerScreen} />
      <Stack.Screen
        name={Screens.SEARCH_RESULTS_SCREEN}
        component={SearchResultsScreen}
      />
      <Stack.Screen name={Screens.FEEDBACK_SCREEN} component={FeedBackScreen} />
    </Stack.Navigator>
  );
};

export default React.memo(AppStack);
