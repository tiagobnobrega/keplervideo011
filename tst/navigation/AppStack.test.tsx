import { createStackNavigator } from '@amzn/react-navigation__stack';
import { render } from '@testing-library/react-native';
import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import AppStack from '../../src/components/navigation/AppStack';
import { Screens } from '../../src/components/navigation/types';

jest.mock('../../src/screens/DetailsScreen', () => 'DetailsScreen');
jest.mock('../../src/screens/PlayerScreen', () => 'PlayerScreen');
jest.mock('../../src/screens/SearchResultsScreen', () => 'SearchResultsScreen');
jest.mock('../../src/components/navigation/AppDrawer', () => 'AppDrawer');

jest.mock('@amzn/react-navigation__stack', () => {
  return {
    createStackNavigator: jest.fn().mockReturnValue({
      Navigator: jest.fn(({ children }) => children),
      Screen: jest.fn(({ children }) => children),
    }),
  };
});
const mockStore = configureMockStore();

describe('AppStack', () => {
  let store: any;
  let mockDispatch: any;
  beforeEach(() => {
    store = mockStore({
      videoDetail: {
        watchList: [],
        purchasedList: [],
        rentList: [],
      },
    });
    mockDispatch = jest.fn();
    (useDispatch as unknown as jest.Mock).mockReturnValue(mockDispatch);
    (useSelector as unknown as jest.Mock).mockImplementation(
      (callback: (state: any) => any) => callback(store.getState()),
    );
  });

  const setup = () => {
    return render(<AppStack />);
  };

  it('renders correctly', () => {
    const { toJSON } = setup();
    const tree = toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders the navigator with the correct screens', () => {
    setup();
    const createStack = createStackNavigator();

    expect(createStack.Navigator).toHaveBeenCalled();
    expect(createStack.Screen).toHaveBeenCalledWith(
      expect.objectContaining({ name: Screens.APP_DRAWER }),
      {},
    );
    expect(createStack.Screen).toHaveBeenCalledWith(
      expect.objectContaining({ name: Screens.DETAILS_SCREEN }),
      {},
    );
    expect(createStack.Screen).toHaveBeenCalledWith(
      expect.objectContaining({ name: Screens.PLAYER_SCREEN }),
      {},
    );
    expect(createStack.Screen).toHaveBeenCalledWith(
      expect.objectContaining({ name: Screens.SEARCH_RESULTS_SCREEN }),
      {},
    );
  });

  test('navigator has correct screen options', () => {
    setup();
    const createStack = createStackNavigator();
    const navigatorProps = (createStack.Navigator as jest.Mock).mock
      .calls[0][0];

    expect(navigatorProps.screenOptions).toEqual({
      headerShown: false,
      animationEnabled: false,
    });
  });
});
