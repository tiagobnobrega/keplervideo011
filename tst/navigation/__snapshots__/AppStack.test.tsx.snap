// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppStack renders correctly 1`] = `null`;
